alg/gdal_vectorx.h

```diff
@@ -0,0 +1,290 @@
+/******************************************************************************
+ * Project:  GDAL Vector abstraction
+ * Purpose:
+ * Author:   Javier Jimenez Shaw
+ *
+ ******************************************************************************
+ * Copyright (c) 2024, Javier Jimenez Shaw
+ *
+ * SPDX-License-Identifier: MIT
+ ****************************************************************************/
+
+#ifndef GDAL_VECTORX_H_INCLUDED
+#define GDAL_VECTORX_H_INCLUDED
+
+/*! @cond Doxygen_Suppress */
```

Comment:
I think this is a leftover of a copy-paste. Should this class be in Doxygen? I think so.

Reply:
As we don't install the file, this is debatable. There should be at least one warning in the class introduction that the file is for GDAL internal use for now